### PR TITLE
Bugfix: Prevent continuous restart, if a dependency wasn't installed

### PR DESCRIPTION
**Pull request by isbm**
### What does this PR do?

Maintenance bugfix.
### What issues does this PR fix or reference?

Prevents minion restart forever if a dependency module wasn't installed by a various reasons.
### Tests written?

No

### CODE REVIEW

File: salt/scripts.py [hunk 1]
```diff
@@ -81,14 +81,14 @@ def suicide_when_without_parent(parent_pid):
     try:
         minion = salt.cli.daemons.Minion()
         minion.start()
+    except (ImportError, SystemExit) as exc:
+        log.error(exc)
```
Comment (cachedout) [inline salt/scripts.py]: We no longer want to set the restart flag to False here if we hit a SystemExit?

Comment (isbm) [inline salt/scripts.py]: @cachedout it is _already_ False. We do not want _set_ it to True. :wink:

Comment (cachedout) [inline salt/scripts.py]: Ah, I didn't read up enough. I do like it being explicit but you're right. Thanks!

**Comment by cachedout** [comment]
Some lint here: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/4067/violations/file/salt/scripts.py/

**Comment by isbm** [comment]
@cachedout This is not _my_ lint problem, as this code is already there. Should I extra-fix this in this PR?

**Comment by cachedout** [comment]
I think it's complaining because L86 is now an ancestor to what you added on L84, so yes, this was caused by your change. If you could please clean it up then we'll get this in. Thanks. :]

**Comment by isbm** [comment]
@cachedout here you go. :wink:

**Pull request merged**